metrics/sinks/hawkular/driver_test.go

```diff
@@ -457,6 +457,106 @@ func TestStoreTimeseries(t *testing.T) {
 	assert.NotEqual(t, ids[0], ids[1])
 }
 
+func TestTags(t *testing.T) {
+	m := &sync.Mutex{}
+	calls := make([]string, 0, 2)
+	// how many times tags have been updated
+	tagsUpdated := 0
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		m.Lock()
+		defer m.Unlock()
+		calls = append(calls, r.RequestURI)
+		w.Header().Set("Content-Type", "application/json")
+
+		defer r.Body.Close()
+		b, err := ioutil.ReadAll(r.Body)
+		assert.NoError(t, err)
+
+		if strings.HasSuffix(r.RequestURI, "/tags") {
+			tags := make(map[string]string)
+			err := json.Unmarshal(b, &tags)
+			assert.NoError(t, err)
+
+			assert.Equal(t, 10, len(tags))
+			assert.Equal(t, "test-label", tags["projectId"])
+			assert.Equal(t, "test-container", tags[core.LabelContainerName.Key])
+			assert.Equal(t, "test-podid", tags[core.LabelPodId.Key])
+			assert.Equal(t, "test-container/test/metric/A", tags["group_id"])
+			assert.Equal(t, "test/metric/A", tags["descriptor_name"])
+			assert.Equal(t, "XYZ", tags[core.LabelResourceID.Key])
+			assert.Equal(t, "bytes", tags["units"])
+
+			assert.Equal(t, "testLabelA:testValueA,testLabelB:testValueB", tags[core.LabelLabels.Key])
```

Comment:
The main reason is for backwards compatibility. We don't want to break things for people who are currently using the "labels" tag.

Another reason is that with the "labels" tag we can use regex on the label names themselves (eg find me metrics which have a label which starts with "foo"), which I don't think we can do if we only have the split labels being stored.